main.go

```diff
@@ -111,9 +113,82 @@ func main() {
 		os.Exit(1)
 	}
 
+	createAWSAdapterConfigIfNotPresent(r)
+
 	setupLog.Info("starting manager")
 	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
 }
+
+func getClient() client.Client {
+	cl, err := client.New(ctrl.GetConfigOrDie(), client.Options{Scheme: scheme})
+	if err != nil {
+		setupLog.Error(err, "unable to create client")
+		os.Exit(1)
+	}
+	return cl
+}
+
+type requiredParams struct {
+	clusterName      string
+	clusterRegion    string
+	adapterName      string
+	adapterNamespace string
+}
+
+func (rp *requiredParams) areAllPresent() bool {
+	return rp.clusterName != "" && rp.clusterRegion != "" && rp.adapterName != "" && rp.adapterNamespace != ""
+}
+
+func createAWSAdapterConfigIfNotPresent(r *controllers.AWSAdapterConfigReconciler) {
+	rp := requiredParams{
+		clusterName:      getClusterName(),
+		clusterRegion:    getClusterRegion(),
+		adapterName:      getAdapterName(),
+		adapterNamespace: getAdapterNamespace(),
+	}
+
+	if rp.areAllPresent() {
+		setupLog.Info("One or more of the required parameters could not be found")
```

Comment:
Maybe, you want to also show the param names clusterName="", clusterRegion="us-west", etc.

Reply:
Done.